ui/packages/catalog-ui-search/src/main/webapp/react-component/container/results-export/results-export.tsx

```diff
@@ -171,9 +171,10 @@ class ResultsExport extends React.Component<Props, State> {
       })
     } else {
       const source = this.state.selectedResults[0].source
-      const metacardId = this.state.selectedResults[0].id
-
-      response = await exportResult(source, metacardId, transformerId)
+      const uriEncodedSource = encodeURIComponent(source)
+      const metacardId: any = this.state.selectedResults[0].id
+      const uriEncodedMetacardId: any = encodeURIComponent(metacardId)
+      response = await exportResult(uriEncodedSource, uriEncodedMetacardId, transformerId)
```

Comment:
✏️ I think this works better if you convert metacardId to a number with `parseInt` (and handle NaN result if it's not an int with `isNaN`), and use `encodeURIComponent` on `transformerId` which is a string. Also the `any` types don't seem correct, the default should be string for these calls.

Reply:
I can't keep `metacardId` as a number since `encodeURIComponent()` requires a string and won't compile unless I use `any`. However, `exportResult` requires the metacard ID to be a number which is why I had to use `any` for `uriEncodedMetacardId` (since `encodeURIComponent()` returns a string). We can't convert the metacard ID string to a number anyway since the ID is typically not a numerical value.

I'll encode the transformer ID too.